src/amp-story-player/amp-story-player-impl.js

```diff
@@ -648,7 +648,21 @@ export class AmpStoryPlayer {
    * @private
    */
   next_() {
-    if (this.currentIdx_ + 1 >= this.stories_.length) {
+    if (
+      this.currentIdx_ + 1 >= this.stories_.length &&
+      this.element_.getAttribute('enable-circular-wrapping') != 'true'
+    ) {
+      return;
+    }
+
+    if (
+      this.currentIdx_ + 1 >= this.stories_.length &&
+      this.element_.getAttribute('enable-circular-wrapping') === 'true'
+    ) {
+      this.currentIdx_ = 0;
+      const currentStory = this.stories_[this.currentIdx_];
+      this.show(currentStory.href);
+      this.signalNavigation_();
```

Comment:
I think we can clean this up by extracting it to a method that `next_` and `previous_` can share. They would only need to pass the index they are trying to go to as an argument.